### ISSUE TITLE
CharRNN has a TypeInitializationException 

### ISSUE THREAD

**Issue by ghost** [losttech/Gradient-Samples#6]
```
System.TypeInitializationException
  HResult=0x80131534
  Message=The type initializer for 'numpy.np' threw an exception.
  Source=LostTech.NumPy
  StackTrace:
   at numpy.np.array(Object object, Object dtype, Boolean copy, String order, Boolean subok, Int32 ndmin)
   at CharRNN.TextLoader.Preprocess(String inputFile, String vocabularyFile, String tensorFile, IEnumerable`1& chars, Dictionary`2& vocabulary) in C:\Users\SnowSnakz\Downloads\Gradient-Samples-master\Gradient-Samples-master\CharRNN\CharRNNUtils.cs:line 52
   at CharRNN.TextLoader..ctor(String dataDir, Int32 batchSize, Int32 seqLength, Encoding encoding) in C:\Users\SnowSnakz\Downloads\Gradient-Samples-master\Gradient-Samples-master\CharRNN\CharRNNUtils.cs:line 35
   at CharRNN.CharRNNProgram.Train(CharRNNTrainingParameters args) in C:\Users\SnowSnakz\Downloads\Gradient-Samples-master\Gradient-Samples-master\CharRNN\CharRNNProgram.cs:line 55
   at CharRNN.CharRNNProgram.<>c.<Main>b__2_0(CharRNNTrainingParameters train) in C:\Users\SnowSnakz\Downloads\Gradient-Samples-master\Gradient-Samples-master\CharRNN\CharRNNProgram.cs:line 27
   at CharRNN.CharRNNProgram.Main(String[] args) in C:\Users\SnowSnakz\Downloads\Gradient-Samples-master\Gradient-Samples-master\CharRNN\CharRNNProgram.cs:line 25

Inner Exception 1:
ArgumentNullException: Value cannot be null.
Parameter name: source
```

This occurs with a fresh download.

**Comment by lostmsu** [comment]
@SnowSnakz this looks like a problem with environment configuration. I suggest to add

```csharp
GradientSetup.EnsureInitialized();
```

after `GradientEngine.UseEnvironmentFromVariable();` in `CharRNNProgram.cs`. It should give more meaningful exception.

P.S. it would be easier to help if you can provide the stack trace for the inner exception too (e.g. for the `ArgumentNullException`).

**Comment by lostmsu** [comment]
@SnowSnakz actually, I think the unclear error has been fixed in the upcoming RC0.

You need Python 3.4+ (3.7 recommended) installed with TensorFlow 1.15.0. See [instructions](https://github.com/losttech/Gradient#install-python-tensorflow).

Alternatively, you can reference prepackaged TensorFlow from [NuGet: LostTech.TensorFlow.Python](https://www.nuget.org/packages/LostTech.TensorFlow.Python/). Disadvantage is that then every TensorFlow project deploys its own runtime, so launching every sample will consume ~1GB of disk space. This is the approach RL-MLAgents sample takes: [relevant code](https://github.com/losttech/Gradient-Samples/blob/ad6dada004da95852b38ed44efdaf37f75dca991/RL-MLAgents/Program.cs#L29).

**Comment by ghost** [comment]
@lostmsu I am using Python 3.6 with `tensorflow==1.15` installed.
I will try running the sample with `GradientSetup.EnsureInitialized();`

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/60198160/84966921-5c870b00-b0d8-11ea-8e40-4848240bc7cc.png)
I am getting the same error, no further details...
```
System.ArgumentNullException
  HResult=0x80004003
  Message=Value cannot be null.
Parameter name: source
  Source=Gradient.Runtime
  StackTrace:
   at Gradient.Tools.OnlyOneOrDefault[T](IEnumerable`1 source)
   at Gradient.GradientEngine.GetEnvironment()
   at Gradient.GradientEngine.Initialize()
   at Gradient.GradientEngine.EnsureInitialized()
   at CharRNN.CharRNNProgram.Main(String[] args) in C:\Users\SnowSnakz\Downloads\Gradient-Samples-master\Gradient-Samples-master\CharRNN\CharRNNProgram.cs:line 23
```

**Comment by ghost** [comment]
Ah, I see that I used `GradientEngine` rather than `GradientSetup`... 
If I use GradientSetup, it starts working
```
This trial version of Gradient will expire in 98 days on Friday, September 25, 2020
using Python: 3.6-X64 at C:\Users\SnowSnakz\AppData\Local\Programs\Python\Python36\
reading text file
```
and it continues fine

**Comment by lostmsu** [comment]
@SnowSnakz yeah, one of them will have to go or be named better. The `GradientEngine` is a lower-level thing, that does not care about presence or lack of TensorFlow, it just picks the Python interpreter with the highest version number. In your case it probably picks something, that does not have TensorFlow in it.

**Comment by lostmsu** [comment]
Created an issue for Gradient to clear this confusion: https://github.com/losttech/Gradient/issues/31
